src/compute/src/logging/compute.rs

```diff
@@ -310,6 +313,14 @@ pub(super) fn construct<A: Allocate + 'static>(
             .as_collection()
             .map(arrangement_heap_datum_to_row);
 
+        let error_count = error_count.as_collection().map(move |datum| {
+            Row::pack_slice(&[
+                Datum::String(&datum.export_id.to_string()),
```

Comment:
We should have a different API that would allow us to not allocate a string here... Not for this PR, let's leave it as follow-up work.

Reply:
How would that look like? Having a dedicated `Datum::GlobalId` variant?

Reply:
No, simpler than that. Something like this:
```
error_count.as_collection().map_batch(move |container| {
	let mut string = String::new();
    container.map(move |datum| {
		string.clear();
		write!(&mut string, "{}", datum.export_id);
		Row::pack_slice(&[Datum::String(&string), ...])
  	})
});
```

Reply:
Ah! But why do we need a new API for that? This should work:

```rust
        let mut export_id = String::new();
        let error_count = error_count.as_collection().map(move |datum| {
            export_id.clear();
            write!(&mut export_id, "{}", datum.export_id).unwrap();
            Row::pack_slice(&[
                Datum::String(&export_id),
                Datum::UInt64(u64::cast_from(worker_id)),
                Datum::Int64(datum.count),
            ])
        });
```

Reply:
Opened a separate PR to make this optimization for every log variant: https://github.com/MaterializeInc/materialize/pull/22236